calc/src/mechanics/gen78.ts

```diff
@@ -455,16 +482,18 @@ export function calculateSMSS(
   // below this until we're inside the loop
   let stabMod = 4096;
   if (attacker.hasType(move.type)) {
-    if (attacker.hasAbility('Adaptability')) {
-      stabMod = 8192;
-      desc.attackerAbility = attacker.ability;
-    } else {
-      stabMod = 6144;
-    }
+    stabMod += 2048;
   } else if (attacker.hasAbility('Protean', 'Libero')) {
-    stabMod = 6144;
+    stabMod += 2048;
     desc.attackerAbility = attacker.ability;
   }
+  if (attacker.teraType === move.type) {
```

Comment:
The stabBoost only gets an additional 4/3 multiplier if the base type of the Pokemon matches the tera type

Reply:
That's true, but I've organized this so that normal STAB and tera STAB stack additively, handling both cases.

If you check the math, I think you'll find it works out.